appcelerator/titanium_mobile
# [TIMOB-24660] Android: Support loading downloaded images from gallery

- Implement support for `content://com.android.providers.downloads.documents` paths

###### TEST CASE
- Download an image on device
- Load the image from the app, it's filesize and filename should show correctly
```XML
<uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE"/>
```
```JS
var w = Ti.UI.createWindow({backgroundColor : 'grey'}),
    b = Titanium.UI.createButton({
        title : 'OPEN GALLERY',
        top : '10dp'
    }),
    iv = Ti.UI.createImageView({
        top : '65dp',
        width : '95%'
    }),
    sp = 'android.permission.READ_EXTERNAL_STORAGE';

function openGallery() {
    Ti.Media.openPhotoGallery({
        mediaTypes : [Ti.Media.MEDIA_TYPE_PHOTO],
        success: function(e) {
            iv.height = e.media.height * (iv.rect.width / e.media.width);
            iv.image = e.media;

            alert('filename: ' + e.media.file.name);
            alert('filesize: ' + e.media.length);
        },
        error: function(e) {
            Ti.API.error(JSON.stringify(e));
        }
    });
}

b.addEventListener('click', function(e) {
    if (Ti.Android.hasPermission(sp)) {
        openGallery();
    } else {
        Ti.Android.requestPermissions([sp], function(e) {
            if (e.success) {
                openGallery();
            } else {
                Ti.API.info('ERROR: ' + e.error);
            }
        });
    }
});

w.add(b);
w.add(iv);

w.open();
```

[JIRA Ticket](https://jira.appcelerator.org/browse/TIMOB-24660)
